test/test_custom_ops.py

```diff
@@ -2434,15 +2454,17 @@ def f(
             f: torch.dtype = torch.float,
             g: torch.dtype = torch.float32,
             h: torch.dtype = torch.int,
+            i: torch.device = torch.device("cpu:0"),
+            j: torch.device = "cpu",
         ) -> Tensor:
-            defaults.extend([a, b, c, d, e, f, g, h])
+            defaults.extend([a, b, c, d, e, f, g, h, i, j])
             return x.clone()
 
         x = torch.randn(3)
         f(x)
         self.assertEqual(
             defaults,
-            [None, 3.14, True, 3, "foo", torch.float, torch.float32, torch.int],
+            [None, 3.14, True, 3, "foo", torch.float, torch.float32, torch.int, torch.device("cpu:0"), "cpu"],
```

Comment:
Er I realized that this may not actually be testing what we want. Can you test that `torch.ops._torch_testing.f.default._schema.arguments[-1]` has a reasonable type? I'm not sure if this should be torch.device or `str`, but I think torch.device is more correct here.

Reply:
I think the type for torch.device should just be `str` based on the following error msg? I've also tried `torch.device`, same error. 

raceback (most recent call last):
  File "/Users/shangdiy/Desktop/pytorch/test.py", line 32, in <module>
    @torch.library.custom_op("mylib::foo", mutates_args={})
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/shangdiy/Desktop/pytorch/torch/_library/custom_ops.py", line 123, in inner
    result = CustomOpDef(namespace, opname, schema_str, fn)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/shangdiy/Desktop/pytorch/torch/_library/custom_ops.py", line 169, in __init__
    self._register_to_dispatcher()
  File "/Users/shangdiy/Desktop/pytorch/torch/_library/custom_ops.py", line 445, in _register_to_dispatcher
    cpp_schema = _C.parse_schema(schema_str)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
**RuntimeError: expected string_literal but found 'ident' here:
foo(Tensor x, Device y=Device(cpu:0)) -> Tensor**

Reply:
I added a test like below. torch.device is correct, but the default values for `f` and `g`, and `h` are weird. They ae 6, 6, 3 instead of `torch.float, torch.float32, torch.int`. Is this expected?
```python
        default_values = [arg.default_value for arg in torch.ops._torch_testing.f.default._schema.arguments]
        print(default_values)
        self.assertEqual(
            default_values,
            [None, None, 3.14, True, 3, "foo", torch.float, torch.float32, torch.int, torch.device("cpu:0"), torch.device("cpu")],
        )
```

Reply:
Note that `infer_schema` output doesn't have 6, 6, 3:

`(Tensor x, SymInt? a=None, float b=3.14, bool c=True, SymInt d=3, str e="foo", ScalarType f=float32, ScalarType g=float32, ScalarType h=int32, Device i="cpu:0", Device j="cpu") -> Tensor`

Reply:
> , but the default values for f and g, and h are weird. They ae 6, 6, 3 instead of torch.float, torch.float32, torch.int. Is this expected?

Unfortunately, yes. We pack dtypes into integers in C++. They never made it back into torch.dtype, but this is fine. Fixing this is BC-breaking and we don't really need to fix it.

> I added a test like below. torch.device is correct, 

That's good, thank you.